rust/kernel/error.rs

```diff
@@ -57,6 +58,26 @@ impl Error {
     pub fn to_kernel_errno(&self) -> c_types::c_int {
         self.0
     }
+
+    /// Creates an [`Error`] from a kernel pointer.
+    ///
+    /// # Pointer value range
+    ///
+    /// According to `include/linux/err.h`:
+    /// - [0, .., `core::usize::MAX - bindings::MAX_ERRNO`) is the range for
+    ///   normal values of pointer.
+    ///
+    /// - [`core::usize::MAX - bindings::MAX_ERRNO`,..,`core::usize::MAX] is the
+    ///   range for error values stored in pointer.
+    pub fn from_kernel_ptr(ptr: *mut c_types::c_void) -> Option<Error> {
+        let value = ptr as usize;
+
+        if value >= core::usize::MAX - bindings::MAX_ERRNO as usize {
```

Comment:
Is this correct? `core::usize::MAX - bindings::MAX_ERRNO as usize` is `-4096`, rather than `-4095`.

```suggestion
        if value >= core::usize::MAX - bindings::MAX_ERRNO as usize + 1 {
```

Reply:
I think you're right. This is indeed an embarrassing bug, and prove the point ;-)